### ISSUE TITLE
Enable Chromium Site Isolation

### ISSUE THREAD

**Issue by amaitland** [cefsharp/CefSharp#2967]
`CEF` recently added support for https://bitbucket.org/chromiumembedded/cef/issues/2498/add-support-for-site-per-process

We should now look at properly supporting the process model as it's now the `CEF` default.

- [x] Remove [process-per-site-instance](https://github.com/cefsharp/CefSharp/blob/master/CefSharp.Core/AbstractCefSettings.h#L48) command line arg from `CefSettings`

Related issue #2553

**Comment by amaitland** [comment]
`WPF` version will require `disable-site-isolation-trials` command line arg until [upstream issue #2789](https://bitbucket.org/chromiumembedded/cef/issues/2789/osr-branch-3904-iframes-doesnt-handle) has been resolved

**Comment by amaitland** [comment]
https://bitbucket.org/chromiumembedded/cef/issues/2814/process-messages-cant-be-sent-to-all

**Comment by amaitland** [comment]
Site isolation trials disabled for all `WPF/WinForms/OffScreen` in commit https://github.com/cefsharp/CefSharp/commit/93336a00312f106b95f627bcd3ea9a7fb3f58d12 Now using the default process model (chromium site isolation project)

**Comment by amaitland** [comment]
> `WPF` version will require `disable-site-isolation-trials` command line arg until [upstream issue #2789](https://bitbucket.org/chromiumembedded/cef/issues/2789/osr-branch-3904-iframes-doesnt-handle) has been resolved

This has been resolved (builds haven't been updated yet)

> https://bitbucket.org/chromiumembedded/cef/issues/2814/process-messages-cant-be-sent-to-all

Cannot remove `disable-site-isolation-trials` until this is solved.

**Comment by amaitland** [comment]
Issue is now tracked at https://bitbucket.org/chromiumembedded/cef/issues/2849/siteisolation-process-message-from-browser

**Comment by amaitland** [comment]
Issue has been resolved see https://bitbucket.org/chromiumembedded/cef/issues/2849/siteisolation-process-message-from-browser#comment-60550396

There are potential changes to how frames are handled that need to be evaluated.

**Comment by amaitland** [comment]
> This means that frame objects (including for the main frame) will now always change after cross-origin navigation, and the old frame object will be invalidated.

Cross origin navigations now invalidate the main frame.

**Comment by amaitland** [comment]
With site isolation enabled many of the tests (pretty much anything that uses the IPC) will fail, so reverting for now and reopening.

**Comment by amaitland** [comment]
Adding this to the `94` milestone as a reminder to revisit, debugging should be done before `Chrome` decides to remove `disable-site-isolation-trials` and we're forced to upgrade.

**Comment by amaitland** [comment]
`Chromium Site Isolation` is now enabled by default in [95.7.140-pre](https://github.com/cefsharp/CefSharp/issues/3866#issuecomment-955950545) which is now available for testing. 

You can still manually disable via the `disable-site-isolation-trials` command line arg. e.g.

```c#
var settings = new CefSettings()
{
	//By default CefSharp will use an in-memory cache, you need to specify a Cache Folder to persist data
	CachePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "CefSharp\\Cache")
};

settings.CefCommandLineArgs.Add("disable-site-isolation-trials");

Cef.Initialize(settings, performDependencyCheck: true, browserProcessHandler: null);
```

> Tests are passing with the exception of IntegrationTestFacts.LoadLegacyJavaScriptBindingQunitTestsSuccessfulCompletion. Same test passes when I run it through Visual Studio Test Explorer. A new issue should be created and debugged at some point to determine the cause and evaluate if it's fixable or if upstream changes are required.

Issue https://github.com/cefsharp/CefSharp/issues/3867 has been created to track investigating the failures. Any problems using `Legacy JavaScript Binding` then you'll need to disable `Chromium Site Isolation`

---

If you are using `CefSharp` in a commercial product and not already a sponsor then please consider [supporting](https://github.com/cefsharp/CefSharp#financial-support) my efforts. Developing/maintaining/supporting the project takes a considerable amount of time.

**Comment by amaitland** [comment]
> Known behaviour changes in CEF are as follows: - A spare renderer process may be created on initial browser creation or cross- origin navigation. This spare process may be used with a future cross-origin navigation or discarded on application shutdown. 

As per https://bitbucket.org/chromiumembedded/cef/issues/2498/add-support-for-site-per-process#comment-54190803. With `Site Isolation` enabled it's expected there is a spare render process is created and will exist until `Cef.Shutdown` has been called.
